quinn/benches/bench.rs

```diff
@@ -0,0 +1,143 @@
+use std::cell::RefCell;
+use std::net::{IpAddr, Ipv6Addr, SocketAddr, UdpSocket};
+use std::rc::Rc;
+
+use criterion::{criterion_group, criterion_main, BatchSize, Benchmark, Criterion, Throughput};
+use futures::{Future, Stream};
+use tokio;
+
+use quinn::{ClientConfigBuilder, Endpoint, NewStream, ServerConfigBuilder};
+
+criterion_group!(benches, throughput);
+criterion_main!(benches);
+
+fn throughput(c: &mut Criterion) {
+    let mut server_config = ServerConfigBuilder::default();
+    let cert = rcgen::generate_simple_self_signed(vec!["localhost".into()]);
+    let key = quinn::PrivateKey::from_der(&cert.serialize_private_key_der()).unwrap();
+    let cert = quinn::Certificate::from_der(&cert.serialize_der()).unwrap();
+    let cert_chain = quinn::CertificateChain::from_certs(vec![cert.clone()]);
+    server_config.certificate(cert_chain, key).unwrap();
+
+    let mut server = Endpoint::new();
+    server.listen(server_config.build());
+    let server_sock = UdpSocket::bind(SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), 0)).unwrap();
+    let server_addr = server_sock.local_addr().unwrap();
+    let (server_driver, _, server_incoming) = server.from_socket(server_sock).unwrap();
+
+    let mut client_config = ClientConfigBuilder::default();
+    client_config.add_certificate_authority(cert).unwrap();
+    let mut client = Endpoint::new();
+    client.default_client_config(client_config.build());
+    let (client_driver, client, _) = client
+        .bind(SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), 0))
+        .unwrap();
+
+    let mut runtime = tokio::runtime::current_thread::Runtime::new().unwrap();
+    runtime.spawn(server_driver.map_err(|e| panic!("server driver failed: {}", e)));
+    runtime.spawn(client_driver.map_err(|e| panic!("client driver failed: {}", e)));
+
+    let runtime = Rc::new(RefCell::new(runtime));
+    {
+        let nested = runtime.clone();
+        runtime
+            .borrow_mut()
+            .spawn(server_incoming.for_each(move |(driver, _, incoming)| {
+                nested.borrow_mut().spawn(driver.map_err(|_| panic!()));
```

Comment:
Replacing this with `tokio::runtime::current_thread::spawn` should do the trick, I think.